### PR TITLE
feat(askvm): add a 'bitwiseAnd' resource (& binary operator)

### PR DESCRIPTION
**Pull request by akeni**

### CODE REVIEW

**Comment by akeni** [comment]
Do you want to throw exceptions if number is bigger than 32 bit?

**Comment by czerwinskilukasz1** [comment]
Thanks for writing the test!  I see that you created it for AskCode, which is the "machine language" used in AskQL internals.
It would be great if you could add also similar tests for `bitwiseAnd` and `&` in AskScript, which is the user-facing language. 
  The best place for them would be in [src/askscript/__tests__/13-real-operators/](https://github.com/xFAANG/askql/tree/master/src/askscript/__tests__/13-real-operators) directory. 
`real-operators-10-equals.test.ask` shows how to write tests in AskScript language itself.
You can use the same examples you wrote in AskCode, just rewrite them to AskScript - one test file for `bitwiseAnd` resource and one for `&` operator to make sure both constructs work. 

> Do you want to throw exceptions if number is bigger than 32 bit?

Great question :) For now, let's skip it for now and just write a note. Could you create a separate Issue for throwing such exceptions for all the bitwise operators?

**Comment by mhagmajer** [comment]
You can use https://github.com/xFAANG/askql/pull/397 as reference

**Comment by czerwinskilukasz1** [comment]
Hello @akeni , what's your current thinking?  Let me know what you need, we need your feedback to improve :)

**Comment by akeni** [comment]
> Hello @akeni , what's your current thinking? Let me know what you need, we need your feedback to improve :)

Hi, sorry i was busy with some other work. Will finish it today :)

**Comment by czerwinskilukasz1** [comment]
Ah, I just noticed you added a new commit.  @akeni , a good way Github introduced for letting the reviewer know the PR is ready for re-review is to click on the circular arrows:
image

**Comment by czerwinskilukasz1** [comment]
It seems the PR has currently a small conflict:
image

Would you mind resolving it?   Let me know if you need any pointers on this.

File: src/askscript/__tests__/09-operators/operators-13-bitwiseAnd-res.test.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  const environment = { resources };
+  expect(await e2e('1:bitwiseAnd(1)', environment)).toBe(1);
+  expect(await e2e('0:bitwiseAnd(1)', environment)).toBe(0);
+  expect(await e2e('1:bitwiseAnd(0)', environment)).toBe(0);
+  expect(await e2e('0:bitwiseAnd(0)', environment)).toBe(0);
+  expect(await e2e('-55:bitwiseAnd(-55)', environment)).toBe(-55);
+  expect(await e2e('10:bitwiseAnd(-10)', environment)).toBe(2);
+  expect(await e2e('2147483647:bitwiseAnd(2147483647)', environment)).toBe(
+    2147483647
+  );
```
Comment (czerwinskilukasz1) [inline src/askscript/__tests__/09-operators/operators-13-bitwiseAnd-res.test.ts:14]: Nice test :) 
Why don't we add it to the oper test too?

Comment (akeni) [reply src/askscript/__tests__/09-operators/operators-13-bitwiseAnd-res.test.ts:14]: okay, will do :)

**Comment by czerwinskilukasz1** [comment]
Looks good to me. Congrats for your PR! :)

**Comment by akeni** [comment]
Thanks for the feedback and patience 👍

**Pull request merged**